public/app/features/explore/Explore.tsx

```diff
@@ -324,6 +325,12 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
     const showRichHistory = openDrawer === ExploreDrawer.RichHistory;
     const showQueryInspector = openDrawer === ExploreDrawer.QueryInspector;
     const showLogsVolume = !!logsVolumeDataProvider;
+    // Only show node graph if X-Ray datasource or node graph is enabled
+    const nodeGraphEnabled =
+      datasourceInstance &&
+      (datasourceInstance?.type === 'grafana-x-ray-datasource' ||
+        (['jaeger', 'zipkin', 'tempo'].includes(datasourceInstance.type) &&
```

Comment:
Do we need to check for `['jaeger', 'zipkin', 'tempo']`? Can 3rd-party plugins implement service maps? If yes, this check will disable service map for them.

Reply:
Good point, other datasources could implement and use the node graph. I'll find a more generic way to check if it's enabled.